src/compiler/emitter.ts

```diff
@@ -6498,7 +6498,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 switch (compilerOptions.jsx) {
                     case JsxEmit.React:
                         let text = trimReactWhitespace(node);
-                        if (text.length === 0) {
+                        if (text === undefined || text.length === 0) {

```

Comment:
Since `""` if falsy, you could also just return `text || undefined`, but this might just be clearer anyway.